.github/workflows/deploy-changed-cf.yaml

```diff
@@ -16,49 +26,87 @@ env:
   region: 'eu-west-1'
 
 jobs:
+  setup:
+    name: Check secrets and set up matrix
+    runs-on: ubuntu-20.04
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - id: check-secrets
+        name: Check if necessary secrets are installed.
+        run: |-
+          echo Checking if secrets are defined in the repository.
+
+          if [ -z "${{ secrets.AWS_ACCESS_KEY_ID }}" ] || [ -z "${{ secrets.AWS_SECRET_ACCESS_KEY }}" ]; then
+              echo "This workflow requires AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY defined in this repository secrets to complete."
+              echo "The secrets can be set/rotated by running 'rotate-ci' from 'biomage-utils'."
+          fi
+
+      - id: set-matrix
+        name: Set up cluster matrix
+        run: |-
+          # Github workflow_dispatch inputs do not use use default value if workflow is not
+          # triggered by dispatch. Therefore, we have to manually provide a default value
+          CLUSTER=${CLUSTER:-"staging"}
+
+          if [ "${GITHUB_REF}" == "refs/heads/master" ] || [ "${CLUSTER}" = "staging and production" ]; then
+            echo '::set-output name=matrix::["staging", "production"]'
+          elif [ "${CLUSTER}" = "staging" ]; then
+            echo '::set-output name=matrix::["staging"]'
+          elif [ "${CLUSTER}" = "production" ]; then
+            echo '::set-output name=matrix::["production"]'
+          fi
+        env:
+          CLUSTER: ${{ github.event.inputs.cluster }}
+          GITHUB_REF: ${{ github.ref }}
+
   get-modified-templates:
     name: Fetch paths to modified CloudFormation templates
     runs-on: ubuntu-20.04
     outputs:
       files: ${{ steps.check-number-of-cf-files.outputs.files }}
       num-files: ${{ steps.check-number-of-cf-files.outputs.num-files }}
     steps:
+      - id: checkout
+        name: Check out current branch source code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
       - id: get-changed-files
         name: Get changed files
-        uses: lots0logs/gh-action-get-changed-files@2.1.4
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          echo "[]" > empty.json
 
-      - id: check-number-of-cf-files
-        name: Check CloudFormation templates
-        run: |-
-          # Merge all modified and renamed files into one JSON array
-          #jq -s add ${HOME}/files_added.json ${HOME}/files_modified.json ${HOME}/files_renamed.json > ${HOME}/files_to_check.json
+          # Get name of changed files
+          # and concatenate into a JSON array
+          git diff --name-only --diff-filter=ACMRT remotes/origin/master ${{ github.sha }} | xargs -I 'value' jq --arg filename "value" '. |= . + [$filename]' empty.json | jq -s 'add' > ${HOME}/files.json
```

Comment:
why did you change this?

Reply:
This is because the action `lots0logs/gh-action-get-changed-files@2.1.4` can only be used for `pull` and `push` actions, meanwhile we want to do this also for `workflow_dispatch` action. This was addressed here: https://github.com/hms-dbmi-cellenics/iac/pull/428#discussion_r896794017.